### PR TITLE
Feature/eip1 2386 map sqs send application to print message

### PR DESCRIPTION
**Pull request by Neil-Massey**

### CODE REVIEW

File: src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/RdsElectoralRegistrationOfficeMapper.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package uk.gov.dluhc.printapi.rds.mapper
+
+import org.mapstruct.Mapper
+import org.mapstruct.Mapping
+import uk.gov.dluhc.printapi.dto.EroManagementApiEroDto
+import uk.gov.dluhc.printapi.rds.entity.ElectoralRegistrationOffice
+
+@Mapper
+interface RdsElectoralRegistrationOfficeMapper {
```
Comment (Neil-Massey) [inline src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/RdsElectoralRegistrationOfficeMapper.kt:9]: Prefix is so Spring Application context can start up without having duplicate bean names.

File: src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateMapper.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    @Mapping(target = "vacNumber", expression = "java( idFactory.vacNumber() )")
+    @Mapping(source = "localAuthority", target = "issuingAuthority")
+    @Mapping(target = "printRequests", expression = "java( toPrintRequestList(message, ero) )")
+    abstract fun toCertificate(
+        message: SendApplicationToPrintMessage,
+        ero: EroManagementApiEroDto,
+        localAuthority: String
+    ): Certificate
+
+    protected fun toPrintRequestList(
```
Comment (Neil-Massey) [inline src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateMapper.kt:32]: Couldn't get MapStruct to do this automagically..

Comment (vishalgupta-valtech) [reply src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateMapper.kt:32]: Try this AfterMapping 

```
@Autowired
    protected lateinit var idFactory: IdFactory

    @Autowired
    protected lateinit var printRequestMapper: PrintRequestMapper

    @Mapping(target = "id", ignore = true)
    @Mapping(target = "vacNumber", expression = "java( idFactory.vacNumber() )")
    @Mapping(source = "localAuthority", target = "issuingAuthority")
    abstract fun toCertificate(
        message: SendApplicationToPrintMessage,
        ero: EroManagementApiEroDto,
        localAuthority: String
    ): Certificate

    @AfterMapping
    protected fun addPrintRequestToCertificate(
        message: SendApplicationToPrintMessage,
        ero: EroManagementApiEroDto,
        @MappingTarget certificate: Certificate
    ) {
        certificate.addPrintRequest(printRequestMapper.toPrintRequest(message, ero))
    }
```

File: src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/InstantMapper.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.dluhc.printapi.rds.mapper
+
+import org.springframework.stereotype.Component
+import java.time.Instant
+import java.time.OffsetDateTime
+
+@Component
+class InstantMapper {
+    fun toInstant(offsetDateTime: OffsetDateTime?): Instant? = offsetDateTime?.toInstant()
+}
```
Comment (vishalgupta-valtech) [inline src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/InstantMapper.kt:10]: Do these mapper belong to `rds` package?

Comment (vishalgupta-valtech) [reply src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/InstantMapper.kt:10]: oh may be these are all rds (mysql) related changes 👍🏼 . Ignore above comment then

Comment (Neil-Massey) [reply src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/InstantMapper.kt:10]: Just in `rds` package to keep it clear at this stage which are used for RDS... a future task will move to correct package 👍

**Review by vishalgupta-valtech — APPROVED** [review]
LGTM, some minor comments/suggestions

File: src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateMapper.kt (hunk 2 above)
Comment (Neil-Massey) [reply src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateMapper.kt:32]: Thanks @vishalgupta-valtech that works, and yes it should be using `add` 👍

Comment (vishalgupta-valtech) [reply src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateMapper.kt:32]: The only difference with using aftermapping is that you can actually call `.addPrintRequest` of Certificate. However IMHO even your impl is ok but just creates a new list of print requests (doesn't add to it)

**Pull request merged**